pywbem_mock/_wbemconnection_mock.py

```diff
@@ -1455,13 +1459,29 @@ def _fake_enumerateclasses(self, namespace, **params):
         Enumerate classes from class repository. If classname parameter
         exists, use it as the starting point for the hiearchy to get subclasses.
 
+        Returns:
+
+            return tuple including list of classes
+
+        Raises:
+
+            CIMError: CIM_ERR_INVALID_NAMESPACE if invalid namespace,
+            CIMError: CIM_ERR_NOT_FOUND if Classname not found
+            CIMError: CIM_ERR_INVALID_CLASS if class that is basis for tes
+                      does not exist
+
         """
         self._get_class_repo(namespace)
 
-        cns = self._get_subclass_names(
-            params.get('classname', None),
-            namespace,
-            params['DeepInheritance'])
+        classname = params.get('ClassName', None)
+        if classname:
+            assert(isinstance(classname, CIMClassName))
+            if not self._class_exists(classname.classname, namespace):
```

Comment:
The added test is correct and the expected error is INVALID_CLASS. However, we go back and to _get_class to get all the classes for the EnumerateClasses and if that methodreturns an error it is CIM_ERR_NOT_FOUND.  
To be completely correct, this would have to be some corruption in the repository since the get_subclass_names just found all those classnames in the same repository but it could still occur but since it is almost impossible  I did not protect against it with a try block around the get_class loop.  That is where the original CIM_ERR_NOT_FOUND exceptions comment came from.